tests/sentry/api/endpoints/test_scim_user_index.py

```diff
@@ -69,6 +71,62 @@ def test_post_users_already_exists(self):
             "detail": "User already exists in the database.",
         }
 
+    def test_post_users_with_role_valid(self):
+        with self.feature({"organizations:scim-orgmember-roles": True}):
+            CREATE_USER_POST_DATA["sentryOrgRole"] = "manager"
+            resp = self.get_success_response(
+                self.organization.slug, method="post", status_code=201, **CREATE_USER_POST_DATA
+            )
+            member = OrganizationMember.objects.get(
+                organization=self.organization, email="test.user@okta.local"
+            )
+
+            correct_post_data = {
+                "schemas": ["urn:ietf:params:scim:schemas:core:2.0:User"],
+                "id": str(member.id),
+                "userName": "test.user@okta.local",
+                "emails": [{"primary": True, "value": "test.user@okta.local", "type": "work"}],
+                "active": True,
+                "name": {"familyName": "N/A", "givenName": "N/A"},
+                "meta": {"resourceType": "User"},
+            }
+            assert correct_post_data == resp.data
+            assert member.role == "manager"
+            member.delete()
+
+            # check role is case insensitive
+            CREATE_USER_POST_DATA["sentryOrgRole"] = "mAnaGer"
+            self.get_success_response(
+                self.organization.slug, method="post", status_code=201, **CREATE_USER_POST_DATA
+            )
+            member = OrganizationMember.objects.get(
+                organization=self.organization, email="test.user@okta.local"
+            )
+            assert member.role == "manager"
+            member.delete()
+
+            # no sentry org role -> default
+            del CREATE_USER_POST_DATA["sentryOrgRole"]
+            self.get_success_response(
+                self.organization.slug, method="post", status_code=201, **CREATE_USER_POST_DATA
+            )
+            member = OrganizationMember.objects.get(
+                organization=self.organization, email="test.user@okta.local"
+            )
+            assert member.role == self.organization.default_role
+
+    def test_post_users_with_role_invalid(self):
+        with self.feature({"organizations:scim-orgmember-roles": True}):
+            # Non-existant role
+            CREATE_USER_POST_DATA["sentryOrgRole"] = "nonexistant"
+            resp = self.get_error_response(
+                self.organization.slug, method="post", status_code=400, **CREATE_USER_POST_DATA
+            )
+            assert resp.data == {
+                "schemas": ["urn:ietf:params:scim:api:messages:2.0:Error"],
+                "detail": "Invalid organization role.",
+            }
+
```

Comment:
I'd like to see at least one test case that returns a `SCIM_400_INVALID_ORGROLE` status, where a user (who otherwise has the right permission to access the endpoint) is stopped from assigning a role above their own.

If I'm wrong about the Manager/Owner thing above and this is impossible under the current role structure, you could force the corner case with `override_settings` on the default role definitions. Or, if we can commit to `SCIM_400_INVALID_ORGROLE` being a logical impossibility, I'd suggest deleting that status and just raising a hard exception (with 500 response) instead.